### PR TITLE
Retry on "Table being modified concurrently" in Delta Lake product tests

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Workaround for #13199

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java [hunk 1]
```diff
@@ -76,6 +77,6 @@ public void testReads()
                 "WITH (location = 'hdfs://hadoop-master:9000/tmp/region')");
 
         assertThat(onTrino().executeQuery("SELECT count(*) FROM delta.default.region")).containsOnly(row(5L));
-        onTrino().executeQuery("DROP TABLE delta.default.region");
+        retryOnModifiedConcurrentlyFailure(() -> onTrino().executeQuery("DROP TABLE delta.default.region"));
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java:80]: As discussed offline.

if this is a situation a user can legitimately run into, retrying in tests will only cover up actual product problem, preventing us from fixing it.

**Review by findepi — CHANGES_REQUESTED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java:80]: That's why I didn't add the retry in `TestDeltaLakeDropTableCompatibility`. I can revert all changes for `onTrino` though.

Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java:80]: > I can revert all changes for `onTrino` though.

👍

Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java]: Reverted.

Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java]: Unfortunately, CI failed when dropping a table on Trino. 
https://github.com/trinodb/trino/actions/runs/3569815668/jobs/6000754659
```
tests               | 	Suppressed: java.lang.Exception: Query: DROP TABLE delta.default.test_compression_GZIP_7lrqddrubi
```

**Pull request closed without merging**